tachyon/crypto/commitments/vector_commitment_scheme.h

```diff
@@ -62,33 +78,44 @@ class VectorCommitmentScheme {
     return derived->DoUnsafeSetup(size, params);
   }
 
-  // Commit to |container| and populates |result| with the commitment.
+  // Commit to |container| and populates |result| with the commitment if
+  // |batch_mode| is false. Otherwise, it stores the commitment in
+  // |batch_commitments_|.
   // Return false if the size of |container| doesn't match with the size of
   // parameters.
   template <typename Container>
-  [[nodiscard]] bool Commit(const Container& container,
-                            Commitment* result) const {
-    const Derived* derived = static_cast<const Derived*>(this);
+  [[nodiscard]] bool Commit(const Container& container, Commitment* result) {
+    Derived* derived = static_cast<Derived*>(this);
+    if constexpr (VectorCommitmentSchemeTraits<Derived>::kSupportsBatchMode) {
+      if (batch_commitment_state_.batch_mode)
+        return derived->DoCommit(container, batch_commitment_state_);
+    }
     return derived->DoCommit(container, result);
   }
 
   // Commit to |container| with a |random_value| and populates |result| with the
-  // commitment. Return false if the size of |container| doesn't match with the
+  // commitment if |batch_mode| is false. Otherwise, it stores the commitment in
+  // |batch_commitments_|.
+  //. Return false if the size of |container| doesn't match with the
```

Comment:
```suggestion
  // Return false if the size of |container| doesn't match with the
```

Reply:
Sorry for such diff. I'll take a closer look when self reviewing :(